File: Exercism/fsharp/proverb/Proverb.fs
module Proverb

let recite (input: string list) =
    let rec reciter =
        function
        | ([], []) -> []
        | (poem, items) when items.Length > 1 ->
            reciter (List.append poem [ $"For want of a {items.[0]} the {items.[1]} was lost." ], List.tail items)
        | (poem, _items) -> List.append poem [ $"And all for the want of a {input.[0]}." ]

    reciter ([], input)

File: Exercism/csharp/proverb/Proverb.cs
using System.Collections.Generic;
using System.Linq;

public static class Proverb
{
    public static string[] Recite(string[] subjects) => subjects.Count() == 0
      ? subjects
      : Reciter(new List<string>(), subjects).Append(string.Format($"And all for the want of a {subjects.First()}.")).ToArray();

    private static IEnumerable<string> Reciter(IEnumerable<string> poem, IEnumerable<string> subjects) => subjects.Count() == 1
      ? poem
      : Reciter(poem.Append($"For want of a {subjects.ElementAt(0)} the {subjects.ElementAt(1)} was lost."), subjects.Skip(1));
}